build.sbt

```diff
@@ -92,24 +92,13 @@ lazy val testStdLayout    = Project("daffodil-test-stdLayout", file("test-stdLay
 lazy val commonSettings = Seq(
   organization := "org.apache.daffodil",
   version := "2.2.0-SNAPSHOT",
-  scalaVersion := "2.11.8",
+  scalaVersion := "2.12.6",
   scalacOptions ++= Seq(
     "-deprecation",
     "-language:experimental.macros",
     "-unchecked",
     "-Xfatal-warnings",
     "-Xxml:-coalescing",
-    "-Ybackend:GenBCode",
-    "-Ydead-code",
-    "-Yinline" ,
-    "-Yinline-warnings",
-    "-Yopt:inline-global",
-    "-Yopt-warnings",
-    "-Ywarn-inaccessible",
-    "-Ywarn-infer-any",
-    "-Ywarn-nullary-override",
-    "-Ywarn-unused",
-    "-Ywarn-unused-import"
```

Comment:
I guess these -Y options are no longer supported in Scala 2.12? Or is 2.12 just better at detecting some warnings? I don't think removing this is a blocker, but we should investigate adding back some of these warnings where it makes sense.